temporian/implementation/numpy/operators/tests/propagate_test.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2021 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from absl.testing import absltest
+
+import numpy as np
+
+from temporian.core.data.sampling import Sampling
+from temporian.core.operators.propagate import Propagate
+from temporian.implementation.numpy.operators.propagate import (
+    PropagateNumpyImplementation,
+)
+from temporian.implementation.numpy.data.event import NumpyEvent, NumpyFeature
+from temporian.implementation.numpy.data.sampling import NumpySampling
+from temporian.core.data import event as event_lib
+from temporian.core.data import feature as feature_lib
+from temporian.core.data import dtype as dtype_lib
+
+
+class PropagateOperatorTest(absltest.TestCase):
+    def setUp(self):
+        pass
+
+    def test_base(self):

```

Comment:
Take a look at how we declared core and numpy events in `calendar_day_of_month_test` using `NumpyEvent.from_dataframe()` and `.schema()`. It would be nice to try and keep tests that test similar stuff (in this case, a numpy implementation) in a similar format. We can discuss on what's the best format in our next meeting 🗣️

Reply:
using .schema() specifically allows to skip the core definitions, which makes it much more concise